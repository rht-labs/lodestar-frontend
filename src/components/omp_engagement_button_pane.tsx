import React, { useContext } from 'react';
import { Button } from '@patternfly/react-core';
import { EngagementFormContext } from '../context/engagement_form_context';
import { EngagementContext } from '../context/engagement_context';
import { slugProperties } from '../utilities/slug_properties';

function _OMPEngagementButtonPane() {
  const engagementContext = useContext(EngagementContext);
  const engagementFormContext = useContext(EngagementFormContext);
  
  const buttonPane: React.CSSProperties = {
    position: 'absolute',
    right: 10,
    bottom: 10,
    backgroundColor: 'rgba(255,255,255, 0.6)',
    padding: 5,
    borderRadius: 4,
  }

  const buttonDisplay: React.CSSProperties = {
    margin: 5,
  }

  const launchCluster = () => {
    engagementContext.createEngagement(
      slugProperties(engagementContext.activeEngagement, [
        'ocp_sub_domain',
        'customer_name',
        'project_name',
      ])
    );
  }

  const saveCluster = () => {
    engagementContext.saveEngagement(engagementFormContext.state);
  }

  return (
    <div style={ buttonPane } >
      <Button onClick={saveCluster} style={ buttonDisplay } > Save </Button>
      <Button isDisabled={true} onClick={launchCluster} style={ buttonDisplay } > Launch </Button>
    </div>
  );
}

export const OMPEngagementButtonPane = React.memo(_OMPEngagementButtonPane);