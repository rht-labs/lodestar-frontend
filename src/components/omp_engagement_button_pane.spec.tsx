import React from 'react';
import { OMPEngagementButtonPane } from './omp_engagement_button_pane';

describe('Engagement Button Pane UI', () => {
  test('should match snapshot', () => {
    expect(<OMPEngagementButtonPane />).toMatchSnapshot();
  });
});
