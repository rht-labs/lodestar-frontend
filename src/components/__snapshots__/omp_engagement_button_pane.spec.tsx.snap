// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Engagement Button Pane UI should match snapshot 1`] = `<UNDEFINED />`;
